tests/publiccloud/aistack_basic.pm

```diff
@@ -0,0 +1,179 @@
+use testapi;
+use serial_terminal 'select_serial_terminal';
+use publiccloud::utils qw(is_byos registercloudguest register_openstack);
+use publiccloud::ssh_interactive 'select_host_console';
+use transactional;
+use containers::k8s;
+use strict;
+use warnings;
+use utils;
+use publiccloud::utils;
+use transactional qw(process_reboot trup_install trup_shell);
+use File::Basename;
+use version_utils;
+use Data::Dumper;
+
+sub tr_run_script {
+    my ($file_name, $ins_cmd) = @_;
+    # Write the script directly
+    script_run("echo '#!/bin/bash' > ~/$file_name");
+    script_run("echo 'set -e' >> ~/$file_name");
+    script_run("echo '$ins_cmd' >> ~/$file_name");
+    script_run("chmod 755 ~/$file_name");
+    script_run("cat ~/$file_name");
+    # Execute the script using trup_run
+    trup_call("run bash -c ~/$file_name");
+    record_info("Executed script $file_name.");
+    # Delete the script
+    script_run("rm ~/$file_name");
+}
+
+sub install_dependency_package {
+    my ($instance, $rke2_url, $kubectl_url, $helm_url) = @_;
+    my $runtime = get_required_var('PUBLIC_CLOUD_AISTACK_DEP_PKG');
+
+    for (split(',\s*', $runtime)) {
+        my $package = $_;
+        my $pkg_status = script_run("rpm -q $package");
+        my $tr_install_cmd = "";
+        record_info("Package status $pkg_status");
+
+        if ($pkg_status == 0) {
+            record_info("$package is already installed");
+        } else {
+            record_info("$package is not installed, installing now...");
+            if ($package eq "rke2-server") {
+                $tr_install_cmd .= "curl -sSL $rke2_url | sudo sh; ";
+            } elsif ($package eq "helm") {
+                $tr_install_cmd .= "[ ! -d /usr/local/bin ] && mkdir -p /usr/local/bin; ";
+                $tr_install_cmd .= "sudo chmod 755 /usr/local/bin; ";
+                $tr_install_cmd .= "curl -sSL $helm_url | sudo sh; ";
+                $tr_install_cmd .= "echo 'export PATH=\$PATH:/usr/local/bin' >> ~/.bashrc; ";
+                $tr_install_cmd .= "source ~/.bashrc; ";
+                $tr_install_cmd .= "helm version; ";
+            } elsif ($package eq "kubectl") {
+                $tr_install_cmd .= "[ ! -d /usr/local/bin ] && mkdir -p /usr/local/bin; ";
+                $tr_install_cmd .= "sudo chmod 755 /usr/local/bin; ";
+                $tr_install_cmd .= "curl -LO $kubectl_url | sudo sh; ";
+                $tr_install_cmd .= "echo 'export PATH=\$PATH:/usr/local/bin' >> ~/.bashrc; ";
+                $tr_install_cmd .= "source ~/.bashrc; ";
+                $tr_install_cmd .= "kubectl version --client; ";
+            }
+            if (($package eq "docker") || ($package eq "curl") || ($package eq "git")) {
+                trup_call("pkg install $package");
+                record_info("Installed $package.");
+            }
+        }    # pkg_status if
+        if (($package eq "rke2-server") || ($package eq "kubectl") || ($package eq "helm")) {
```

Comment:
this One should be and `else` in above trup `if`, 
only option is command for trup-call: 
either a string pkg or a script name, but endin in the same trup-call, followed by reboot, info then RM file if exists.